extension/src/popup/helpers/useSetupSigningFlow.ts

```diff
@@ -55,14 +55,14 @@ export function useSetupSigningFlow(
   };
 
   const signAndClose = async () => {
-    emitMetric(METRIC_NAMES.approveSign);
     if (isHardwareWallet) {
       dispatch(
         startHwSign({ transactionXDR: transactionXdr, shouldSubmit: false }),
       );
       setStartedHwSign(true);
     } else {
       await dispatch(signFn());
+      await emitMetric(METRIC_NAMES.approveSign);
```

Comment:
Nice, I think this makes sense. I wonder what the UX is like if `emitMetric` takes a long time though. I don't think it will typically but in that case the user might experience a hung popup feeling because there isn't a loading state associated with `emitMetric`. 
Would this edge case be handled by emitting the metric from the action itself so that it can piggy back on that loading/success/error lifecycle?

Reply:
Good thought. If you check the video, the loading spinner does continue to spin on the `Sign` button while metric is being emitted to give the visual cue that something is still happening.

The signed xdr is transmitted immediately, so if for some odd reason the emit metric takes a realllllly long time, the dapp already has the xdr and the user can continue whatever they're doing. The popup will just continue to operate and close whenever it finishes